src/main/java/org/spongepowered/api/entity/living/player/Player.java

```diff
@@ -259,4 +261,13 @@ default GameModeData getGameModeData() {
      */
     void setSleepingIgnored(boolean sleepingIgnored);
 
+    /**
+     * Gets all the {@link KeyBinding}s that are available for this {@link Player},
+     * if the player doesn't have a client that supports custom key bindings this
+     * will always return a empty collection.
+     *
+     * @return The key bindings
+     */
+    Collection<KeyBinding> getKeyBindings();
```

Comment:
@phroa The list will never be empty if a player has a client side mod, then will it always contain the default key bindings, since you cannot remove them.
And there will be a forge impl once its updated to 1.9,  I also made a standalone client mod for liteloader (link can be found at the top)